Added python cases for pretty_print_generic

emjin opened the pull request:
Added cases necessary to python function calls, formatting them in the pythonic style.

Tested with

from metrics import send
def foo():
   bar()
   bar + send('my-report-id') + bar()
   (hi, my)
   (hi, my, bye)
   A[1]
   A[-1]
   A[1:4]
   A[1:4:-1]
   A[::-1]
   A[1:]
   if 1 == 2: foo()
   true
   3.3
   foo(self.data)

emjin committed:
Added python cases for pretty_print_generic

Added cases necessary to python function calls, formatting them in the pythonic style.

Tested with

from metrics import send
def foo():
   bar()
   bar + send('my-report-id') + bar()
   (hi, my)
   (hi, my, bye)
   A[1]
   A[-1]
   A[1:4]
   A[1:4:-1]
   A[::-1]
   A[1:]
   if 1 == 2: foo()
   true
   3.3
   foo(self.data)

emjin committed:
Added python cases for pretty_print_generic

Added cases necessary to python function calls, formatting them in the pythonic style.

Tested with

from metrics import send
def foo():
   bar()
   bar + send('my-report-id') + bar()
   (hi, my)
   (hi, my, bye)
   A[1]
   A[-1]
   A[1:4]
   A[1:4:-1]
   A[::-1]
   A[1:]
   if 1 == 2: foo()
   true
   3.3
   foo(self.data)

emjin committed:
Include entire path for imported ents

Tested against

from metrics import send
def foo():
   send('my-report-id')

emjin committed:
Include entire path for imported ents

Tested against

from metrics import send
def foo():
   send('my-report-id')

aryx reviewed (commented):
looks good, not sure about those CI failures.
I had pbs myself.

brendongo committed:
Release 0.12.0

emjin committed:
Linked pattern synthesis through python

Added synthesize-patterns argument, had it call semgrep-core with appropriate arguments

Tested with
python -m semgrep --synthesize-patterns 4:9-4:29 /Users/emma/workspace/semgrep/semgrep-core/tests/SYNTHESIZING/mytest.py

(where mytest.py contains the code referenced in the first commit)

Also ran
semgrep-core -synthesize_patterns 17:3-17:36 tests/SYNTHESIZING/mytest.py

emjin committed:
Linked pattern synthesis through python

Added synthesize-patterns argument, had it call semgrep-core with appropriate arguments

Tested with
python -m semgrep --synthesize-patterns 4:9-4:29 /Users/emma/workspace/semgrep/semgrep-core/tests/SYNTHESIZING/mytest.py

(where mytest.py contains the code referenced in the first commit)

Also ran
semgrep-core -synthesize_patterns 17:3-17:36 tests/SYNTHESIZING/mytest.py

emjin committed:
Merge branch 'develop' into EJ/pattern_from_code

brendongo committed:
fixup! Release 0.12.0

emjin committed:
Merge branch 'develop' into EJ/pattern_from_code

mschwager committed:
More Bento and pre-commit cleanup (#1111)

* More Bento and pre-commit cleanup

* Use pre-commit GHA v2.0.0 which includes automatic caching

* Remove pre-commit requirements.txt checker, we have no requirements.txt

* Use relative import in tests

* Add __init__ files so we can do relative imports

brendongo committed:
Merge pull request #1119 from returntocorp/release-0.12.0

Release 0.12.0

brendongo commented on semgrep/semgrep/synthesize_patterns.py:
```diff
@@ -0,0 +1,26 @@
+def synthesize_patterns(
+    language: str, code_to_synthesize: str, targets_str: List[str]
+) -> None:
+    targets = semgrep.config_resolver.resolve_targets(targets_str)
```
We should probably raise an exception if targets is more than one file.

See dump_ast.py

brendongo commented on semgrep/semgrep/synthesize_patterns.py:
```diff
@@ -0,0 +1,26 @@
+
+    args = ["-synthesize_patterns", code_to_synthesize]
+
+    cmd = [SEMGREP_PATH] + args + targets_str
```
Should use the target that has been resolved in line 15 instead of the user arg.

brendongo commented on semgrep/semgrep/cli.py:
```diff
@@ -191,6 +192,9 @@ def cli() -> None:
             "(can use --json)."
         ),
     )
+    output.add_argument(
```
Can we hide this from help text for now. See --pre-commit argument above.

brendongo reviewed (changes requested):
Would also be nice to add an e2e test for this new command

rcoh committed:
Build from source for tox tests (#1114)

A future diff could cache the semgrep-core artifact based on a hash of the semgrep-core directory to speed up tests in the common case.

mschwager committed:
Avoid monkeypatching PYTHONPATH (#1124)

* Avoid monkeypatching PYTHONPATH

Now that pipenv installs semgrep in editable mode we can avoid
monkeypatching the PYTHONPATH env variable

* Add __init__ file to make directory a module for relative import

rcoh committed:
Don't try to parse an inline pattern as YAML (#1098)

* Don't try to parse an inline pattern as YAML

* Update error message & tests

emjin committed:
Added an e2e test for the synthesize_patterns action

Modified synthesize_patterns to print a json object. Ran with pipenv run pytest tests/e2e/test_synthesize_patterns.py

emjin committed:
Merge branch 'EJ/pattern_from_code' of github.com:returntocorp/semgrep into EJ/pattern_from_code

emjin committed:
Made changes according to pr

aryx commented:
Bento errors seem legit:

Please fix these issues, or:

     > semgrep/semgrep/synthesize_patterns.py:2
     ╷
    2│   import tempfile
     ╵
     = 'tempfile' imported but unused

flake8 unused-module https://lintlyci.github.io/Flake8Rules/rules/F401.html
     > semgrep/semgrep/synthesize_patterns.py:4
     ╷
    4│   from typing import Optional
     ╵
     = 'typing.Optional' imported but unused

aryx commented:
There's lots of merge already in this diff. Let's merge that and you can fix the bento error and brendon suggestion in another PR.

brendongo commented on .github/workflows/env-tests.yml:
```diff
@@ -6,28 +6,60 @@ on:
     branches: [master, develop]
 
 jobs:
+  build-core:
```
These should not have been in the PR

brendongo commented:
There were many files that should not have been in the PR. It should not have been merged like this.

aryx commented:
What did we do wrong? I thought emma just did git merge develop on her branch. The
Only CI error was in bento. 

Sent from my iPhone

> On 25 Jun 2020, at 19:32, Brendon Go <notifications@github.com> wrote:
> 
> ﻿
> There were many files that should not have been in the PR. It should not have been merged like this.
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or unsubscribe.

brendongo commented:
Not sure what happened but it is not immediately clear from the diff what this PR includes.
